omnigibson/prims/rigid_prim.py

```diff
@@ -283,12 +283,15 @@ def set_linear_velocity(self, velocity):
         """
         self._rigid_prim_view.set_linear_velocities(velocity[None, :])
 
-    def get_linear_velocity(self):
+    def get_linear_velocity(self, clone=True):
         """
+        Args:
+            clone (bool): Whether to clone the internal buffer or not when grabbing data
+
         Returns:
             np.ndarray: current linear velocity of the the rigid prim. Shape (3,).
         """
-        return self._rigid_prim_view.get_linear_velocities()[0]
+        return self._rigid_prim_view.get_linear_velocities(clone=clone)[0]
```

Comment:
Here and below, is this cloning really a major source of slowdowns? What happens when we don't clone, is the underlying tensor returned in a mutable way or something?

What I would do here is take these clones out & leave it as it was prior, even if it's slow. I will follow up your PR with one that saves and dumps everything from the physx state using a global rigid prim view and articulation view and whatnot.

Reply:
Yeah...not sure why. I'm guessing it's because we keep cloning different tensors many many times.

For now, the behavior is pretty much consistent all across the repo (`clone=True` by default), except in the context of dumping state. Can we keep this for now for the sake of demo collection, and simply remove the flag once you have your PR ready? I'm guessing the follow-up PR will be nontrivial, so it'd be ideal to not block demo collection with this.

Reply:
resolved in person -- keep for now, revert later